### PR TITLE
Supabase py

### PR DESCRIPTION
**Pull request by samrose**
## Description of changes

added supabase-py python client for supabase and it's dependencies

https://github.com/supabase-community/supabase-py
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/supabase_auth/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, pkgs
+, python3Packages
+, fetchPypi
+, poetry-core
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "gotrue";
```
Comment (risicle) [inline pkgs/development/python-modules/supabase_auth/default.nix:9]: ?

**Comment by risicle** [comment]
This should be one commit per added package, with the commit message as specified in `CONTRIBUTING.md`.

If the tests aren't in the pypi packages, have you tried pulling the source straight from github? Otherwise, what with these packages also not having `pythonImportsCheck`, they _will_ get broken without anyone noticing.

File: pkgs/development/python-modules/supabase_auth/default.nix (hunk 1 above)
Comment (samrose) [reply pkgs/development/python-modules/supabase_auth/default.nix:9]: @risicle it's had a recent name change, but I guess we'd want to use the name as found pypi, so I'll revise. Thanks!

**Comment by samrose** [comment]
> This should be one commit per added package, with the commit message as specified in `CONTRIBUTING.md`.
> 
> If the tests aren't in the pypi packages, have you tried pulling the source straight from github? Otherwise, what with these packages also not having `pythonImportsCheck`, they _will_ get broken without anyone noticing.

Thanks, I'll revise that per your feedback.

**Comment by dddictionary** [comment]
Hi, I am in need of this package to be added into nixpkgs. Is there anything I can do to help push this?

**Comment by Siegema** [comment]
@dddictionary 
I'm trying to get this cleared because I needed it as well
https://github.com/NixOS/nixpkgs/pull/398562

**Pull request closed without merging**